### DISCUSSION TITLE
A slideshow edition

### DISCUSSION THREAD

**Discussion by kookma** [TiddlyWiki/TiddlyWiki5#5248] [Ideas]
Using Tiddlywiki for slideshow and presentation specially in these days with high demand of e-learning and virtual classes in universities and schools will be very useful.

There are great tools out there like **Introduction** edition, **TalkyTalky**, **Tiddlyshow**, **SlidenStories**, **TiddlyText**, ...
But having a simple official slideshow edition is very demanding. This way, TW may use Javascript, PageTemplate, Layout
as standard elements to create a simple to use slideshow edition.

Recently I have started Tamash plugin, as rewrite of Tiddlyshow and I will be happy to help for an official slideshow edition.

**References**
- https://dynalist.io/d/zUP-nIWu2FFoXH-oM7L7d9DM#q=slide
- https://dynalist.io/d/zUP-nIWu2FFoXH-oM7L7d9DM#q=tiddlyshow
- https://github.com/kookma/TW-Tamasha (will be released after Tiddlywiki 5.1.23)
- https://github.com/sukima/tiddlywiki-reveal-js/ 
- https://andjar.github.io/tiddlytext/

**Comment by Jermolene** [comment]
Hi @kookma a challenge we have at the moment is that editions are published as part of the core, and so we can't update them outside of core releases. That makes that route less attractive for editions that are undergoing rapid development.

I'm not familiar with all the candidates you mention, but another issue is perhaps that the fact that we've got so many slideshow editions might reflect the fact that there are many slightly different, maybe conflicting needs out there. It may well be that we need an ecosystem of slideshow editions to cover all the use cases we encounter.

**Reply by TiddlyTweeter** [reply]
@Jermolene replied to @kookma ...
> we've got so many slideshow editions might reflect the fact that there are many slightly different, maybe conflicting needs out there. It may well be that we need an ecosystem of slideshow editions to cover all the use cases we encounter.

This is a most interesting thread! It highlights a central issue. That issue is: is it better to foreground ONE (using a neologism) _canonical edition_ over multiple alternatives?

---

My thinking here is that from an end-user (beginner) viewpoint is it may be best to deliver a full-solution (i.e. AN edition) on a "slide-show" as the first priority. Sure it could have _links_ to alternative implementations ... But overall I find the idea of having ONE solution upfront is ultimately better.

**Reply by kookma** [reply]
@TiddlyTweeter - I have created a light presentation plugin (much lighter and more native than Tiddlyshow) will be released soon. It is called Tamasha, see below pots. By the way I think an edition (needs a theme) would be great.

**Reply by TiddlyTweeter** [reply]
> @TiddlyTweeter - I have created a light presentation plugin (much lighter and more native than Tiddlyshow) will be released soon. It is called Tamasha, see below pots. By the way I think an edition (needs a theme) would be great.

@kookma thanks for the update note. I'll look at it as soon as its ready.

**Comment by kookma** [comment]
@Jermolene - Yes there are few slideshow editions with similar features, Tiddlyshow and Tiddlywiki-Reveal.js are the most flexible and easy to start ones!

But I asked for a simple like Introduction to be used by most users, so is it possible to improve the current Introduction edition or you encourage third party plugins?

**Reply by Jermolene** [reply]
> But I asked for a simple like Introduction to be used by most users, so is it possible to improve the current Introduction edition or you encourage third party plugins?

I'm very open to improvements to the "Introduction" edition, but we have yet to have any third-party plugins in the official library because of the release issue I mentioned above.

**Reply by silvyn** [reply]
Hi @kookma just post URL about https://github.com/sukima/tiddlywiki-reveal-js/ if someone search it 😄

**Reply by kookma** [reply]
@silvyn - thank you! I edited the first post and added  https://github.com/sukima/tiddlywiki-reveal-js/ to references.

**Reply by TiddlyTweeter** [reply]
@kookma wrote ...
> @Jermolene - Yes there are few slideshow editions with similar features, Tiddlyshow and Tiddlywiki-Reveal.js are the most flexible and easy to start ones!

Practically speaking "TiddlyShow" is elegant and easy to use.  Regarding "Tiddlywiki-Reveal.js" it's biggest plus is that it adds (visually) an "up & down" navigator to the horizontal one (i.e. "last/next"). I doubt most end users would use the "up/down."

Just a comment. TT

**Comment by saqimtiaz** [comment]
@kookma completely offtopic but I just wanted to say that I quite enjoy the persian inspired names of your plugins. Tamasha!

**Reply by kookma** [reply]
@Saq -  Tamasha" means "To follow an interesting process or entertainment. But you may know that
<div dir=rtl>
در لغتنامه معنای تماشا:
نگریستن چیزهایی که سرور آورد یا موجب دلتنگی و اندوه شود و عبرت آورد و هرچیز حیرت انگیز که موجب تعجب و شگفتی باشد و هرچیز که دارای سود و فایده بود
من به معنای دیدن برای آموختن و لذت بردن یا عبرت گرفتن آوردم.
</div>

**Reply by saqimtiaz** [reply]
@kookma yes it is a creative but appropriate name!
